### PR TITLE
Track iter progress

### PR DESCRIPTION
**Pull request by ZwwWayne**
Currently, the progress bar only supports functions & tasks, but not support iterators, which is also a common use of the progress bar, so I add two functions, track_iter_progress & track_enum_progress that could track the progress of iterations in for loop.

This branch has been reformated with yapf & pep8, after adding unitest, it passed the local code coverage test.

### CODE REVIEW

File: docs/utils.md [hunk 1]
```diff
@@ -54,6 +54,25 @@ mmcv.track_parallel_progress(func, tasks, 8)  # 8 workers
 
 ![progress](_static/parallel_progress.gif)
 
+What is more, if you want to iterate or enumerate a list of items and track the progress, `track_iter_progress`
```
Comment (hellock) [inline docs/utils.md]: `What is more` may be unnecessary.

File: mmcv/utils/progressbar.py [hunk 2]
```diff
@@ -172,3 +172,35 @@ def track_parallel_progress(func,
...
+
+    Tasks are done with a simple for-loop.
+
+    Args:
+        func (callable): The function to be applied to each task.
+        tasks (list or tuple[Iterable, int]): A list of tasks or
+            (tasks, total num).
+        bar_width (int): Width of progress bar.
+
+    Returns:
```
Comment (hellock) [inline mmcv/utils/progressbar.py]: `Yields` instead of `Returns`.

File: mmcv/utils/progressbar.py [hunk 3]
```diff
@@ -172,3 +172,35 @@ def track_parallel_progress(func,
...
     return results
+
+
+def track_iter_progress(tasks, bar_width=50, **kwargs):
+    """Track the progress of tasks execution with a progress bar.
+
+    Tasks are done with a simple for-loop.
+
+    Args:
+        func (callable): The function to be applied to each task.
```
Comment (hellock) [inline mmcv/utils/progressbar.py]: Incorrect argument.

Comment (ZwwWayne) [reply mmcv/utils/progressbar.py]: Sorry for missing these, I have updated a new version of the docs and code comments.

**Pull request merged**